pkg/resources/view_grant.go

```diff
@@ -139,6 +144,9 @@ func CreateViewGrant(d *schema.ResourceData, meta interface{}) error {
 	if _, ok := d.GetOk("view_name"); ok {
 		viewName = d.Get("view_name").(string)
 	}
+	if err := d.Set("view_name", viewName); err != nil {
+		return err
+	}
 	var schemaName string
 	if _, ok := d.GetOk("schema_name"); ok {
 		schemaName = d.Get("schema_name").(string)
```

Comment:
Is this block still needed? Or would it be enough to just use?
```suggestion
viewName := d.Get("view_name").(string)
schemaName := d.Get("schema_name").(string)
```
Especially this seems odd:
```
	if _, ok := d.GetOk("view_name"); ok {
		viewName = d.Get("view_name").(string)
	}
	if err := d.Set("view_name", viewName); err != nil {
		return err
	}
```

I don't know, if this is special to this grant, I just have picked this grant for reviewing/testing.

Reply:
I agree this doesn't make a lot of sense, its more to uphold existing contract. if `on_future` or `on_all` is set to true  then "view_name" (or "table_name", "materialized_view_name", etc.) is expected to be an empty string. We can either remove that expectation, or add in a default value to the object name. I am thinking removing that  expectation is sufficient.

Reply:
I think you're right @Relativity74205, good catch! @sfc-gh-swinkler what was the intent for this block? From what I can tell the only difference is that it would now set view_name to an empty string if it wasn't set by the user, but I assume that's already the default value?